### ISSUE TITLE
imu_link to base_link offset removal

### ISSUE THREAD

**Issue by jcaraban** [cra-ros-pkg/robot_localization#700]
I am suspicious of the way **navsat_transform::imuCallback** and **ros_filter::preparePose** correct the IMU mounting offset.

See https://github.com/cra-ros-pkg/robot_localization/blob/noetic-devel/src/navsat_transform.cpp#L714
and https://github.com/cra-ros-pkg/robot_localization/blob/noetic-devel/src/ros_filter.cpp#L2787

It gives me the impression that **navsat_transform** meant to use `yaw_offset_` the second time, but due to poor naming the same `yaw_offset` is corrected for twice. Then the code was perhaps copied to **ros_filter** and the same bug was introduced, even though there is no yaw_offset_ param there.

Btw, I think the quaternion alternative would be `imu_orientation = imu_orientation * imu2base.getRotation().inverse()`

_edit:_ upon further inspection, the current code seems to work as expected 🤔 ... I don't fully understand why, but other users might stumble upon this logic, so please consider the quaternion alternative 👍

**Comment by ayrton04** [comment]
Yeah, the quaternion approach would be better, for sure. Feel free to submit a PR.

I'm not clear on where you're seeing the correction happening twice? In `navsat_transform_node`, `yaw_offset_` is meant to correct for the difference between where the IMU reads 0 radians and where the ENU frame considers 0 radians to be.

In the second case, we're correcting the IMU's orientation. In that case, `yaw_offset` is just the yaw component of the IMU's mounting orientation. The naming convention is indeed poor (and would be solved by the quaternion approach above), but I'd want to see some tests to confirm that the resulting output would be the same. I think doing that would change what users need to specify for the `yaw_offset` parameter.
